reason-react-tutorial.md

```diff
@@ -0,0 +1,681 @@
+
+To make our component template we'll call the `ReasonReact.statelessComponent` function. Passing `"Card"` as the argument gives our component its' name.
+
+{{< highlight javascript >}}
+let component = ReasonReact.statelessComponent("Card");
+{{< /highlight >}}
+
+To actually create a component using our template we need to define a function with the name of `make`. This `make` function takes our component's **props** as labelled arguments (a labelled argument in Reason is an argument starting with a `~`).
+
+For our use cases, we'll have our `Card` component use `name`, `description` and an `href` props. This will give us enough to see what repos match our search as well as include links to them!
+
+{{< highlight javascript >}}
+let make = (~name, ~description, ~href, _children) => {};
+{{< /highlight >}}
+
+In addition, the `make` function has to take a `children` argument as its' last argument, even if the component doesn't do anything with its' children! This is to preserve the type-safety of ReasonReact so that it can do all of its' compile-time magic later on. If you don't plan on using the `children` prop, just add an underscore to the beginning (`_children`) to tell the compiler you didn't plan on using the variable.
+
+Now that we've got an empty `make` function, what should it return? ReasonReact expects `make` to return a record with a bunch of internal keys and the component's lifecycle hooks. Fortunately, we can use that template we made earlier! Let's spread the template into our `make` function's return value.
+
+{{< highlight javascript hl_lines="2">}}
+let make = (~name, ~description, ~href, _children) => {
+  ...component,
+};
+{{< /highlight >}}
+
+It's also worth noting that if you're coming from JavaScript land, curly braces after an arrow in Reason don't behave like curly braces in JavaScript. In Reason, the curly braces after the arrow mean we're actually returning a record, as opposed to just starting a new multiline function body.
+
+Now that we've spread all of our template into `make`, let's add our own custom `render` function. 
+
+{{< highlight javascript >}}
+let make = (~name, ~description, ~href, _children) => {
+  ...component,
+  render: _self =>
+    <div>
+      <h3>
+        <a href target="_blank" rel="noopener noreferrer">
+          {ReasonReact.string(name)}
+        </a>
+      </h3>
+      <p> {ReasonReact.string(description)} </p>
+    </div>,
+};
+{{< /highlight >}}
+
+Let's take a quick look at the JSX first. It's built-in to Reason at the language level but you might notice a few differences from the JSX you're used to.
+
+First off, Reason JSX supports _punning_&mdash;when the prop name matches the variable that's being passed as its' value, you can just write the prop once instead of twice! So since we already a variable named `href` we don't need to write `make={make}` when applying it to our `<a>` tag. Instead, we can just do `<a href>`.
```

Comment:
for italic use * instead of _

Reply:
Something I added recently: https://github.com/rwieruch/blog_robinwieruch_content/blob/master/README.md

Reply:
Sweet...thanks for this link! I'll go through and make sure the formatting is right 👌